scripts/enforce-version-bump

```diff
@@ -10,6 +10,6 @@ last_tag=$(git describe --tags --abbrev=0)
 last_tag_version=${last_tag#v}
 current_version=$(extract_version_from_current_wd)
 
-if [[ $(printf "$current_version\n$last_tag_version" | sort --version-sort | tail -n 1) == "$last_tag_version" ]]; then
```

Comment:
Out of curiosity, what's the problem with printf?

Reply:
The same problem is with C `printf`, or even Java `String.format`: non-constant values should **not** occur in the format string, as they might include `%...` format characters (which in this case, won't be printed at all, as there're no further params). Not a serious issue here, but shellcheck is right in general